pkg/teeth/publish_logs.go

```diff
@@ -18,74 +18,31 @@
 package teeth
 
 import (
-	"bytes"
 	"context"
 	"fmt"
-	"text/template"
-	"time"
-
-	"cloud.google.com/go/bigquery"
-
-	_ "embed"
 )
 
-// PublisherSourceQuery is the source query that teeth job pipeline
-// will use to publish results.
-//
-//go:embed sql/publisher_source.sql
-var PublisherSourceQuery string
-
 // BigQueryClient defines the spec for calls to read from and write to
 // BigQuery tables.
 type BigQueryClient interface {
-	Config() *BQConfig
-	Query(string) *bigquery.Query
-}
-
-// TODO: Add query limit param.
-//
-// BQConfig defines configuration parameters for the BigQuery tables
-// used by the teeth job pipeline.
-type BQConfig struct {
-	PullRequestEventsTable       string
-	InvocationCommentStatusTable string
-	EventsTable                  string
-	LeechStatusTable             string
+	QueryLatest(context.Context) ([]*PublisherSourceRecord, error)
+	Insert(context.Context, []*InvocationCommentStatusRecord) error
 }
 
-// PublisherSourceRecord maps the columns from the driving BigQuery query
-// to a usable structure.
-type PublisherSourceRecord struct {
-	DeliveryID     string    `bigquery:"delivery_id"`
-	PullRequestID  int       `bigquery:"pull_request_id"`
-	PullRequestURL string    `bigquery:"pull_request_html_url"`
-	Received       time.Time `bigquery:"received"`
-	LogsURI        string    `bigquery:"logs_uri"`
-	HeadSHA        string    `bigquery:"head_sha"`
-}
-
-// InvocationCommentStatusRecord is the output data structure that maps to the
-// teeth pipeline's output table schema for invocation comment statuses.
-type InvocationCommentStatusRecord struct {
-	PullRequestID  int                `bigquery:"pull_request_id"`
-	PullRequestURL string             `bigquery:"pull_request_html_url"`
-	ProcessedAt    time.Time          `bigquery:"processed_at"`
-	CommentID      bigquery.NullInt64 `bigquery:"comment_id"`
-	Status         string             `bigquery:"status"`
-	JobName        string             `bigquery:"job_name"`
-}
-
-// SetUpPublisherSourceQuery converts the PublisherSourceQuery string into a
-// Query object that the BigQueryClient can then execute. It populates the
-// query parameters with the BigQuery config values.
-func SetUpPublisherSourceQuery(ctx context.Context, bqClient BigQueryClient) (*bigquery.Query, error) {
-	tmpl, err := template.New("publisher").Parse(PublisherSourceQuery)
+// GetLatestSourceRecords gets the latest publisher source records.
+func GetLatestSourceRecords(ctx context.Context, bqClient BigQueryClient) ([]*PublisherSourceRecord, error) {
```

Comment:
This func and the one below - I can't tell if they are necessary. They seem to be making a simple call the the bqClient.

Reply:
It serves as an abstraction layer between the (yet-to-be-implemented) handler and bigquery client. Note that the bigquery client is passed in as an interface. This allows the handler tests to stub this client as a test double later on.

Reply:
Why can't you use the bigqueryclient interface as the abstraction layer? If you plan to add more logic here, then I can agree it's useful. Otherwise, I still don't understand the purpose.